.github/workflows/hardfork_test.yml

```diff
@@ -0,0 +1,108 @@
+name: Hardfork test
+
+on:
+  push:
+  pull_request:
+  merge_group:
+  workflow_dispatch:
+
+jobs:
+  hardfork-test:
+    strategy:
+      matrix:
+        # Supported GitHub-hosted runners and hardware resources
+        # see https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources
+        os: [ubuntu-22.04]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Cache of Cargo
+      uses: actions/cache@v3
+      with:
+        path: |
+          ~/.cargo/bin/
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
+        key: ${{ matrix.os }}-${{ runner.os }}-${{ runner.arch }}-cargo-build-${{ hashFiles('**/Cargo.lock') }}
+        restore-keys: |
+          ${{ matrix.os }}-${{ runner.os }}-${{ runner.arch }}-cargo-build
+    - name: Build Axon in the development profile
+      run: cargo build
+
+    - name: Start multiple Axon nodes
+      env:
+        LOG_PATH: ${{ runner.temp }}/${{ matrix.os }}/multi-axon-nodes
+      run: |
+        echo "LOG_PATH: ${{ env.LOG_PATH }}"
+        mkdir -p ${{ env.LOG_PATH }}
+
+        target/debug/axon --version
+
+        sed -i 's/hardforks = \[\]/hardforks = ["None"]/g' devtools/chain/specs/multi_nodes/chain-spec.toml
+        grep "hardforks" devtools/chain/specs/multi_nodes/chain-spec.toml
+
+        for id in 1 2 3 4; do
+          target/debug/axon init \
+            --config     devtools/chain/nodes/node_${id}.toml \
+            --chain-spec devtools/chain/specs/multi_nodes/chain-spec.toml \
+            > ${{ env.LOG_PATH }}/node_${id}.log
+        done
+
+        for id in 1 2 3 4; do
+          target/debug/axon run \
+            --config     devtools/chain/nodes/node_${id}.toml \
+            >> ${{ env.LOG_PATH }}/node_${id}.log &
+        done
+
+        npx zx <<'EOF'
+        import { waitXBlocksPassed } from './devtools/ci/scripts/helper.js'
+        await Promise.all([
+          waitXBlocksPassed('http://127.0.0.1:8001', 4),
+          waitXBlocksPassed('http://127.0.0.1:8002', 4),
+          waitXBlocksPassed('http://127.0.0.1:8003', 4),
+          waitXBlocksPassed('http://127.0.0.1:8004', 4),
+        ])
+        EOF
+      timeout-minutes: 1
+
+    - name: Checkout sunchengzhu/axon-hardfork-test
+      uses: actions/checkout@v4
+      with:
+        repository: sunchengzhu/axon-hardfork-test
+        path: axon-hardfork-test
```

Comment:
Discussion:
In another PR, shall we move the code of `axon-hardfork-test` in to the https://github.com/sunchengzhu/axon/tree/847745d34e77cdf9c2e7f0f05ffdec17571aaeeb/examples dir?